### PR TITLE
amfaller test dump

### PR DESCRIPTION
**Pull request by amfaller**
Hi,

I wanted to make my tests available. I believe they should all be ready to use as-is, i.e. the `make check` rule from previous given makefiles should work if these tests are just copied into a `project_4/tests` directory. If any of them don't seem good, though, please feel free to take those ones out.

Thanks,
Tony Faller

### CODE REVIEW

File: tls_tests/src/test_clone_read.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+int tls_create(unsigned int size);
+int tls_destroy();
+int tls_read(unsigned int offset, unsigned int length, char *buffer);
+int tls_write(unsigned int offset, unsigned int length, const char *buffer);
+int tls_clone(pthread_t tid);
+
+int ret;
+char *msg = "Hello";
+int pthread_id;
```
Comment (dannosliwcd) [inline tls_tests/src/test_clone_read.c]: This should be type `pthread_t`. Depending on the underlying data type of `pthread_t`, storing as an int may result in both width and sign changes when stored as an int and later cast back.

File: tls_tests/src/test_clone_read.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+void *threadFn(void *id_ptr){
+    int thread_id = *(int*)id_ptr;
+    
+    pthread_id = (int)pthread_self();
```
Comment (dannosliwcd) [inline tls_tests/src/test_clone_read.c]: No casts between pthread_t and int should be needed

File: tls_tests/src/test_clone_read.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    pthread_t myThread;
+    int myInt = 1;
+    char buffer[10] = {0};
+    
+    pthread_create(&myThread, NULL, threadFn, &myInt);
+    
+    sleep(1);
+    
+    printf("Main: Trying to clone thread %d...\n", myInt);
+    ret = tls_clone((pthread_t)pthread_id);
```
Comment (dannosliwcd) [inline tls_tests/src/test_clone_read.c]: Similarly, no casts should be needed here

File: tls_tests/src/test_clone_write_read.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+int tls_create(unsigned int size);
+int tls_destroy();
+int tls_read(unsigned int offset, unsigned int length, char *buffer);
+int tls_write(unsigned int offset, unsigned int length, const char *buffer);
+int tls_clone(pthread_t tid);
+
+int ret;
+char *msg = "Hello";
+int pthread_id;
```
Comment (dannosliwcd) [inline tls_tests/src/test_clone_write_read.c]: Same comments about int and pthread_d as in the other clone test.

File: tls_tests/src/test_write_read_multipage_1.c [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    assert( strcmp(buffer, "Hello Worl") == 0 );
+    printf("...Main: Read successful. Received message: %s\n\n", buffer);   // Expect "Hello Worl"
+    
+    printf("Main: Trying to destroy my TLS...\n");
+    ret = tls_destroy();
+    assert(ret == 0);
+    printf("...Main: Destroy successful\n");
+
+    return (EXIT_SUCCESS);
+}
```
Comment (dannosliwcd) [inline tls_tests/src/test_write_read_multipage_1.c]: Please end this file with a newline. Some compilers might complain about missing newlines (at least clang does).

File: tls_tests/src/test_EC_create_2.c [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    assert(ret == 0);
+    printf("...Main: Obtained\n\n");
+
+    printf("Main: Trying to obtain another TLS...\n");
+    ret = tls_create(100);
+    assert(ret == -1);
+    printf("...Main: Unable to obtain\n");
+    
+    return (EXIT_SUCCESS);
+}
```
Comment (dannosliwcd) [inline tls_tests/src/test_EC_create_2.c]: Please end this file with a newline.

**Review by dannosliwcd — CHANGES_REQUESTED** [review]
Thanks for sharing! I like how the tests contain combinations of high-level comments describing their purpose, plus progress printfs that give context in the event of a test failure.

There are some comments requesting some minor changes.

**Comment by amfaller** [comment]
I've gone through and implemented the changes you've described -- please let me know if you can see them and if I did them right. This is my first time doing a pull request outside of my own repos ^_^

**Comment by dannosliwcd** [comment]
> This is my first time doing a pull request outside of my own repos ^_^

Awesome! Looks good to me, and looks like you have a handle on PRs!

**Pull request merged**